

<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.5" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<title>Yet another SuSiE benchmark</title>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<style type="text/css">
  div.input_prompt {display: none;}
  div.output_html {
     font-family: "PT Mono", monospace;
     font-size: 10.0pt;
     color: #353535;
     padding-bottom: 25px;
 }
  pre:not([class]) {
    background-color: white;
  }
</style>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'] ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>

</head>

<body>
<style type = "text/css">
@font-face {
 font-family: 'Droid Sans';
 font-weight: normal;
 font-style: normal;
 src: local('Droid Sans'), url('fonts/droid-sans.ttf') format('truetype');
}
@font-face {
 font-family: 'Fira Code';
 font-weight: normal;
 font-style: normal;
 src: local('Fira Code'), url('fonts/firacode.otf') format('opentype');
}
@font-face {
 font-family: 'PT Mono';
 font-weight: normal;
 font-style: normal;
 src: local('PT Mono'), url('fonts/ptmono.ttf') format('truetype');
}

body {
  font-family: "Droid Sans";
  font-size: 160%;
  padding-top: 66px;
  padding-bottom: 40px;
}

h1, h2, h3, h4, h5, h6 {
  margin-top: 20px;
 }

a.anchor-link:link {
  text-decoration: none;
  padding: 0px 20px;
  visibility: hidden;
}

h1:hover .anchor-link,
h2:hover .anchor-link,
h3:hover .anchor-link,
h4:hover .anchor-link,
h5:hover .anchor-link,
h6:hover .anchor-link {
  visibility: hidden;
}

.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Yet another SuSiE benchmark</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="./index.html">Overview</a>
</li>
        
<li>
  <a href="./analysis.html">Analysis</a>
</li>
        
      </ul>
    
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/KaiqianZhang/dsc-susie"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="SuSiE-demonstrated-in-DSC">SuSiE demonstrated in DSC<a class="anchor-link" href="#SuSiE-demonstrated-in-DSC">&#182;</a></h1><h2 id="Goal">Goal<a class="anchor-link" href="#Goal">&#182;</a></h2><p>This is a Dynamic Statistical Comparison for testing SuSiE performance given a design matrix <code>X</code> from real data, e.g. GTEx, single cell, or GSEA data.</p>
<p>Our goal is to investigate how SuSiE performs (see "Performance evaluation criteria" below) in the context of different X structures.</p>
<h2 id="Data">Data<a class="anchor-link" href="#Data">&#182;</a></h2><h3 id="GTEx-gene-expression-data">GTEx gene expression data<a class="anchor-link" href="#GTEx-gene-expression-data">&#182;</a></h3><p>Gene expression from two tissues: whole blood and brain cells. Design matrix <code>X</code> is a gene expression matrix from 755 whole blood and 439 brain cell samples on 1000 genes.</p>
<h3 id="Single-cell-gene-expression-data">Single cell gene expression data<a class="anchor-link" href="#Single-cell-gene-expression-data">&#182;</a></h3><p>Gene expression of two types of hard-to-distinguished cells. Design matrix <code>X</code> has 600 regulatory T cells and 600 memory T cells on 1000 genes.</p>
<h3 id="GSEA-gene-set-data">GSEA gene-set data<a class="anchor-link" href="#GSEA-gene-set-data">&#182;</a></h3><p>Gene-set / pathway annotation of genes. Design matrix <code>X</code> has 1200 genes on 1000 pathways. Each column is a gene-set / pathway of binary coding where <code>1</code> indicates the corresponding row (gene) is in the gene-set, <code>0</code> otherwise.</p>
<h3 id="Random-data">Random data<a class="anchor-link" href="#Random-data">&#182;</a></h3><p>Random data are a 1200 by 1000 matrix, each entry of which is randomly generated from <code>rnorm(0,1)</code>.</p>
<h3 id="GTEx-genotype-data">GTEx genotype data<a class="anchor-link" href="#GTEx-genotype-data">&#182;</a></h3><p>Design matrix <code>X</code> is a 574 by 1001 matrix.</p>
<h2 id="Simulation">Simulation<a class="anchor-link" href="#Simulation">&#182;</a></h2><p>We simulate 50 datasets. For each dataset, we simulate a gaussian <code>y</code> under various number of causal variables and total percentage of variance explained (PVE).</p>
<h2 id="Analysis-methods">Analysis methods<a class="anchor-link" href="#Analysis-methods">&#182;</a></h2><p>We run SuSiE to investigate various questions of interest. For instance, performance on different dataset, factors that impact power, choice of priors, and so on. See "Investigations" section below.</p>
<h2 id="Performance-evaluation-criteria">Performance evaluation criteria<a class="anchor-link" href="#Performance-evaluation-criteria">&#182;</a></h2><p>We evaluate SuSiE performance by the following four numerical values:</p>
<ul>
<li><strong>Power</strong>: number of confidence sets (CS) that contain at least one true effect, divided by number of true effects</li>
<li><p><strong>FDR</strong>: an estimate for false discovery rate, 1 - (number of CS that contain at least one true effect devided by number of CS reported)</p>
</li>
<li><p><strong>CS number</strong>: number of CS reported</p>
</li>
<li><p><strong>CS size</strong>: median size of CS reported</p>
</li>
<li><p><strong>CS purity</strong>: average "purity" of CS, an indication of correlation of variables within CS</p>
</li>
<li><p><strong>top hit rate</strong>: define "top hit" as a true effect that receives a SuSiE posterior inclusion probability (PIP) being the highest PIP among all variables in the same reported CS. Top hit rate is then given by the number of top hit divided by the number of CS that contains at least one true effect.</p>
</li>
</ul>
<h2 id="Investigations">Investigations<a class="anchor-link" href="#Investigations">&#182;</a></h2><h3 id="GTEx-gene-expression-data">GTEx gene expression data<a class="anchor-link" href="#GTEx-gene-expression-data">&#182;</a></h3><ul>
<li><p>Power of SuSiE</p>
<ul>
<li><a href="analysis/GTEX_gene_expression_analysis_1.html">one non-zero effects</a></li>
<li><a href="analysis/GTEX_gene_expression_analysis_2.html">multiple non-zero effects</a></li>
</ul>
</li>
<li><p>Choice of prior</p>
<ul>
<li><a href="analysis/GTEX_gene_expression_analysis_3.html">one non-zero effect</a></li>
<li><a href="analysis/GTEX_gene_expression_analysis_4.html">two non-zero effects</a></li>
</ul>
</li>
<li><p>Large number of causal variables</p>
<ul>
<li><a href="analysis/GTEX_gene_expression_analysis_5.html">performance of 200 non-zero effects</a></li>
</ul>
</li>
<li><p>Different initialization methods</p>
</li>
</ul>
<h3 id="Single-cell-data">Single cell data<a class="anchor-link" href="#Single-cell-data">&#182;</a></h3><ul>
<li><p>Power of SuSiE</p>
<ul>
<li><a href="analysis/Singe_cell_analysis_1.html">one non-zero effects</a></li>
<li><a href="analysis/Singe_cell_analysis_2.html">multiple non-zero effects</a></li>
</ul>
</li>
<li><p>Choice of prior</p>
<ul>
<li><a href="analysis/Singe_cell_analysis_3.html">one non-zero effect</a></li>
<li><a href="analysis/Singe_cell_analysis_4.html">two non-zero effects</a></li>
</ul>
</li>
<li><p>Large number of causal variables</p>
<ul>
<li><a href="analysis/Singe_cell_analysis_5.html">performance of 200 non-zero effects</a></li>
</ul>
</li>
<li><p>Different initialization methods</p>
</li>
</ul>
<h2 id="DSC-instructions">DSC instructions<a class="anchor-link" href="#DSC-instructions">&#182;</a></h2><h3 id="General-instructions">General instructions<a class="anchor-link" href="#General-instructions">&#182;</a></h3><p>The main DSC file is <code>benchmarks.dsc</code>. To see what is available:</p>

<pre><code>./benchmark.dsc -h</code></pre>
<p>before running,</p>

<pre><code>rm -Rf benchmark.scripts.html benchmark.html benchmark</code></pre>
<p>and to run the benchmark for 50 datasets:</p>

<pre><code>dsc benchmark.dsc --replicate 50</code></pre>
<p>Or to run a minimal test benchmark, e.g.</p>

<pre><code>dsc benchmark.dsc --truncate</code></pre>
<p>To run on the cluster:</p>

<pre><code>dsc benchmark.dsc --host midway.yml --replicate 50 -c 40</code></pre>
<h3 id="Input-X">Input <code>X</code><a class="anchor-link" href="#Input-X">&#182;</a></h3><p>To input a customized design matrix <code>X</code>, please go to <code>benchmark.dsc</code>. At the bottom, change the <code>pathX</code> to the path of your matrix <code>X</code> as you want.</p>

</div>
</div>
</div>
<hr>
&copy 2018 Kaiqian Zhang and Gao Wang at Stephens Lab, University of Chicago
<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

</div>

<script>
// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
